cmds/boot/boot.go

```diff
@@ -133,107 +100,171 @@ func getSupportedFilesystem() ([]string, error) {
 
 }
 
-// mountEntry tries to mount a specific block device
-func mountEntry(path string, supportedFilesystem []string) (bool, error) {
-	var returnValue bool
+// mountEntry tries to mount a specific block device using a list of
+// supported file systems. We have to try to mount the device
+// itself, since devices can be filesystem formatted but not
+// partitioned; and all its partitions.
+func mountEntry(d string, supportedFilesystem []string) error {
 	var err error
-	exist, err := os.Stat("/u-root")
-	if exist == nil {
-		err = syscall.Mkdir("/u-root", 0777)
-		if err != nil {
-			return false, err
-		}
+	verbose("Try to mount %v", d)
+
+	// find or create the mountpoint.
+	m := filepath.Join(uroot, d)
+	if _, err = os.Stat(m); err != nil && os.IsNotExist(err) {
+		err = os.MkdirAll(m, 0777)
 	}
-	var flags uintptr
-	// Was supposed to be unecessary for kernel 4.x.x
-	if verbose {
-		log.Printf("/dev/" + path)
+	if err != nil {
+		verbose("Can't make %v", m)
+		return err
 	}
 	for _, filesystem := range supportedFilesystem {
-		flags = syscall.MS_MGC_VAL
-		// Need to load the filesystem kind supported
-		exist, err = os.Stat("/u-root/" + path)
-		if exist == nil {
-			err = syscall.Mkdir("/u-root/"+path, 0777)
-			if err != nil {
-				return false, err
-			}
-		}
-		err := syscall.Mount(filepath.Join("/dev/",path), filepath.Join("/u-root/",path), filesystem, flags, "")
-		if err == nil {
-			return true, nil
+		// Was supposed to be unecessary for kernel 4.x.x
+		var flags = uintptr(syscall.MS_MGC_VAL | syscall.MS_RDONLY)
```

Comment:
that magic val is still unnecessary :) has been since 2.4. if it fails, something else is wrong.